Site/admin/components/Account/MergeConfirm.php

```diff
@@ -0,0 +1,258 @@
+/**
+ * Account merge confirmation page
+ *
+ * @package   Site
+ * @copyright 2016 silverorange
+ */
+abstract class SiteAccountMergeConfirm extends AdminDBConfirmation
+{
+	// {{{ protected properties
+
+	/**
+	 * @var integer
+	 */
+	protected $id;
+
+	/**
+	 * @var SiteAccount
+	 */
+	protected $account;
+
+	/**
+	 * @var integer
+	 */
+	protected $id2;
+
+	/**
+	 * @var SiteAccount
+	 */
+	protected $account2;
+
+	/**
+	 * @var boolean
+	 */
+	protected $keep_first;
+
+	// }}}
+
+	// init phase
+	// {{{ protected function initInternal()
+
+	protected function initInternal()
+	{
+		parent::initInternal();
+
+		$this->id = SiteApplication::initVar('id');
+		$this->account = $this->getAccount($this->id, $this->account);
+
+		$this->id2 = SiteApplication::initVar('id2');
+		$this->account2 = $this->getAccount($this->id2, $this->account2);
+
+		$this->keep_first = SiteApplication::initVar('keep_first');
+	}
+
+	// }}}
+	// {{{ protected function getAccount()
+
+	protected function getAccount($id, $account)
+	{
+		if ($account === null) {
+			$account_class = SwatDBClassMap::get('SiteAccount');
+
+			$account = new $account_class();
+			$account->setDatabase($this->app->db);
+
+			if (!$account->load($id)) {
+				throw new AdminNotFoundException(sprintf(
+					Site::_('An account with an id of ‘%d’ does not exist.'),
+					$id
+				));
+			}
+
+			$instance_id = $this->app->getInstanceId();
+			if ($instance_id !== null) {
+				if ($account->instance->id !== $instance_id) {
+					throw new AdminNotFoundException(sprintf(
+						Store::_('Incorrect instance for account ‘%d’.'),
+						$id
+					));
+				}
+			}
+		}
+
+		return $account;
+	}
+	// }}}
+	// {{{ protected function getSourceAccount()
+	protected function getSourceAccount()
+	{
+		if (!$this->keep_first) {
+			return $this->account;
+		}
+		return $this->account2;
+	}
+	// }}}
+	// {{{ protected function getTargetAccount()
+	protected function getTargetAccount()
+	{
+		if ($this->keep_first) {
+			return $this->account;
+		}
+		return $this->account2;
+	}
+	// }}}
+
+	// process phase
+	// {{{ protected function processDBData()
+	protected function processDBData()
+	{
+		parent::processDBData();
+
+		$source_account = $this->getSourceAccount();
+		$target_account = $this->getTargetAccount();
+
+		$transaction = new SwatDBTransaction($this->app->db);
+		try {
+			$this->mergeAccounts($source_account, $target_account);
+
+			$now = new SwatDate();
+			$now->toUTC();
+			$source_account->delete_date = $this->app->db->quote($now, 'date');
+			$this->addNote($source_account, $target_account);
+
+			$target_account->delete_date = null;
+			$this->addNote($target_account, $source_account);
+
+			$transaction->commit();
+		} catch (Exception $e) {
+			$transaction->rollback();
+			var_dump($e);
+			throw $e;
+		}
+
+		$message = new SwatMessage(
+			sprintf(
+				Site::_(
+					'Successfully merged the account "%s" into "%s".'
+				),
+				SwatString::minimizeEntities($source_account->email),
```

Comment:
minimizeEntities (HTML safe) is only required if we explicitly set the message content type to `text/xml` otherwise it happens automatically to the whole message.